components/wifi/esp8266-driver/ESP8266Interface.cpp

```diff
@@ -187,18 +188,19 @@ void ESP8266Interface::_connect_async()
         _cmutex.unlock();
         return;
     }
-
-    if (_esp.connect(ap_ssid, ap_pass) != NSAPI_ERROR_OK) {
+    _connect_retval = _esp.connect(ap_ssid, ap_pass);
+    if (_connect_retval == NSAPI_ERROR_OK || _connect_retval == NSAPI_ERROR_AUTH_FAILURE
+            || _connect_retval == NSAPI_ERROR_CONNECTION_TIMEOUT || _connect_retval == NSAPI_ERROR_NO_SSID) {
```

Comment:
I think we should continue trying with NSAPI_ERROR_CONNECTION_TIMEOUT.

Reply:
You are right, although checking this led me to a sad conclusion, that ESP does not realize that "aaaaaaaa" network does not exist and it never returns with "3: cannot find the target AP" error code, as the documentation promises.
Instead after two minutes it returns a "2: wrong password", which we later translate to AUTH_FAILURE.
I even logged into ruka and loaded a cliapp to run "wlan0 scan" to make double sure "aaaaaaaa" does not exist. It doesn't.
To make up for the 2 minute wait I will also extend the wifi testsuite duration.

Reply:
Sounds like a good idea.

Reply:
There is also influence from previous tests - when I run the whole suite the ESP will never return anything else than 1 in WIFI-CONNECT-SECURE-FAIL. I am investigating.